examples/worklet.rs

```diff
@@ -1,161 +1,58 @@
-    channel_config: ChannelConfig,
-    /// audio param controlling the volume (for educational purpose, use a GainNode otherwise)
-    amplitude: AudioParam,
-}
-
-// implement required methods for AudioNode trait
-impl AudioNode for WhiteNoiseNode {
-    fn registration(&self) -> &AudioContextRegistration {
-        &self.registration
-    }
-
-    fn channel_config(&self) -> &ChannelConfig {
-        &self.channel_config
-    }
-
-    // source nodes take no input
-    fn number_of_inputs(&self) -> usize {
-        0
-    }
-
-    // emit a single output
-    fn number_of_outputs(&self) -> usize {
-        1
-    }
-}
-
-impl WhiteNoiseNode {
-    /// Construct a new WhiteNoiseNode
-    fn new<C: BaseAudioContext>(context: &C) -> Self {
-        context.register(move |registration| {
-            // setup the amplitude audio param
-            let param_opts = AudioParamDescriptor {
-                min_value: 0.,
-                max_value: 1.,
+use web_audio_api::node::worklet::{
+    AudioParamValues, AudioWorkletNode, AudioWorkletNodeOptions, AudioWorkletProcessor,
+};
+use web_audio_api::node::{AudioNode, AudioScheduledSourceNode};
+
+use web_audio_api::{AudioParamDescriptor, AutomationRate};
+
+struct MyProcessor;
+
+impl AudioWorkletProcessor for MyProcessor {
+    fn parameter_descriptors() -> Vec<(String, AudioParamDescriptor)>
+    where
+        Self: Sized,
+    {
+        vec![(
+            String::from("gain"),
+            AudioParamDescriptor {
+                min_value: f32::MIN,
+                max_value: f32::MAX,
                 default_value: 1.,
                 automation_rate: AutomationRate::A,
-            };
-            let (param, proc) = context.create_audio_param(param_opts, &registration);
-
-            // setup the processor, this will run in the render thread
-            let render = WhiteNoiseProcessor {
-                amplitude: proc,
-                color: NoiseColor::White,
-            };
-
-            // setup the audio node, this will live in the control thread (user facing)
-            let node = WhiteNoiseNode {
-                registration,
-                channel_config: ChannelConfig::default(),
-                amplitude: param,
-            };
-
-            (node, Box::new(render))
-        })
-    }
-
-    /// The Amplitude AudioParam
-    fn amplitude(&self) -> &AudioParam {
-        &self.amplitude
+            },
+        )]
     }
 
-    fn set_noise_color(&self, color: NoiseColor) {
-        self.registration.post_message(color);
-    }
-}
-
-struct WhiteNoiseProcessor {
-    amplitude: AudioParamId,
-    color: NoiseColor,
-}
-
-impl AudioProcessor for WhiteNoiseProcessor {
-    fn process(
+    fn process<'a, 'b>(
         &mut self,
-        _inputs: &[AudioRenderQuantum],
-        outputs: &mut [AudioRenderQuantum],
-        params: AudioParamValues<'_>,
-        _scope: &RenderScope,
+        inputs: &'b [&'a [f32]],
```

Comment:
These different lifetimes are necessary unfortunately

Reply:
A bit odd/complex indeed, but not really shocking neither in my opinion as this is boilerplate code... 

Do you think it is due to the fact that you need to deal with the existing `AudioWorkletRenderer` so this could potentially be removed once (and if) all nodes are ported to this new API, or is it something we will have to live with forever (with an additional depth when several inputs/outputs are supported) in any case?